numpy/_core/tests/test_multithreading.py

```diff
@@ -271,3 +271,19 @@ def closure(b):
         # Reducing the number of threads means the test doesn't trigger the
         # bug. Better to skip on some platforms than add a useless test.
         pytest.skip("Couldn't spawn enough threads to run the test")
+
+@pytest.mark.parametrize("dtype", [bool, int, float])
+def test_nonzero_bool(dtype):
+    # np.nonzero uses np.count_nonzero to determine the size of the output array
+    # In a second pass the indices of the non-zero elements are determined, but they can have changed
```

Comment:
Maybe add that this test triggers a known data race and the behavior is undefined, we just want to make sure numpy doesn't seg fault.